includes/compat/trait-wc-stripe-subscriptions.php

```diff
@@ -784,36 +784,38 @@ public function maybe_render_subscription_payment_method( $payment_method_to_dis
 
 		$payment_method_to_display = __( 'N/A', 'woocommerce-gateway-stripe' );
 
-		// Retrieve all possible payment methods for subscriptions.
 		try {
-			$sources = array_merge(
-				$stripe_customer->get_payment_methods( 'card' ),
-				$stripe_customer->get_payment_methods( 'sepa_debit' ),
-				$stripe_customer->get_payment_methods( 'cashapp' )
-			);
+			// Retrieve all possible payment methods for subscriptions.
+			foreach ( WC_Stripe_Customer::STRIPE_PAYMENT_METHODS as $payment_method_type ) {
+				foreach ( $stripe_customer->get_payment_methods( $payment_method_type ) as $source ) {
+					if ( $source->id !== $stripe_source_id ) {
+						continue;
+					}
 
-			if ( $sources ) {
-				foreach ( $sources as $source ) {
-					if ( $source->id === $stripe_source_id ) {
-						$card = false;
-						if ( isset( $source->type ) && 'card' === $source->type ) {
-							$card = $source->card;
-						} elseif ( isset( $source->object ) && 'card' === $source->object ) {
-							$card = $source;
-						}
+					$card = false;
 
-						if ( $card ) {
-							/* translators: 1) card brand 2) last 4 digits */
-							$payment_method_to_display = sprintf( __( 'Via %1$s card ending in %2$s', 'woocommerce-gateway-stripe' ), ( isset( $card->brand ) ? $card->brand : __( 'N/A', 'woocommerce-gateway-stripe' ) ), $card->last4 );
-						} elseif ( ! empty( $source->sepa_debit ) ) {
-							/* translators: 1) last 4 digits of SEPA Direct Debit */
-							$payment_method_to_display = sprintf( __( 'Via SEPA Direct Debit ending in %1$s', 'woocommerce-gateway-stripe' ), $source->sepa_debit->last4 );
-						} elseif ( ! empty( $source->cashapp ) ) {
-							/* translators: 1) Cash App Cashtag */
-							$payment_method_to_display = sprintf( __( 'Via Cash App Pay (%1$s)', 'woocommerce-gateway-stripe' ), $source->cashapp->cashtag );
-						}
+					if ( isset( $source->type ) && 'card' === $source->type ) {
+						$card = $source->card;
+					} elseif ( isset( $source->object ) && 'card' === $source->object ) {
+						$card = $source;
+					}
 
-						break;
+					if ( $card ) {
```

Comment:
I think this can be another `switch` clause (`switch ( true )`).

Reply:
`switch ( true )` statements can be a little janky in my opinion and sometimes aren't intuitive. I agree that this if else stack would be better served by a switch statement so I've changed it to a `switch ( $type )` one.